### DISCUSSION TITLE
Server does not support secure connections

### DISCUSSION THREAD

**Discussion by hswamy** [postalserver/postal#1570] [Help with using Postal]
Hello, 
I recently setup postal on ubuntu server, Went through the documentation and configured as it is. Now when i try to test an email i get an error saying : Server does not support secure connections, If i use -UseSSL in below command.

`Send-MailMessage -from 'no-reply@example.com' -to 'aaa.bbb@gmail.com' -Subject 'Test' -Body 'test 1' -SmtpServer 'x.x.x.x' -Port 587 -UseSsl -Credential (get-credential)`

If i dont use -UseSSL, email flows as expected. So my question is how do i enable secure connections.

Any help would be appreciated

**Comment by hswamy** [comment]
FYI:

I have configured port: 587 as well

**Comment by willpower232** [comment]
Have you definitely enabled it and added a certificate?

https://github.com/postalserver/postal/blob/d808d15c28d1c4c626af44bedc05743b01e078b9/config/postal.defaults.yml#L64-L66

**Reply by hswamy** [reply]
yes, I have enabled and added the certificate.
![image](https://user-images.githubusercontent.com/43389515/131547770-c6a08e2a-7753-4e9d-9f59-c49fc9fcf3cf.png)

I used postal log to check what's happening and i found out that, there was an error saying as below .

![image](https://user-images.githubusercontent.com/43389515/131547898-daba4020-860b-474a-acee-911b6cf90ed0.png)
 
My certs are in /opt/postal/config/postal.crt and /opt/postal/config/postal.key
Any help would be appreciated

**Reply by hswamy** [reply]
I gave read permission to .key and .crt file, 

But still if i use -UseSSL, i get an error saying `Send-MailMessage : The remote certificate is invalid according to the validation procedure`

**Reply by willpower232** [reply]
if you run `postal bash app` then you can get into the container and do `ls -hal` on the `/config` directory in the container then you should see how accessible the certificate files are in there

**Reply by XenyFord** [reply]
YES! Mr. Will, I tried it as you said. But nothing worked out for me.

**Reply by XenyFord** [reply]
I didn't find any exact path for these certificate and keys

**Reply by willpower232** [reply]
to go over what you need to do...

you need to create the certificate and key then put them in `/opt/postal/config` so that you have `/opt/postal/config/whatevercertificate.crt` and `/opt/postal/config/whatevercertificate.key`

then you need to set the config to match but without `/opt/postal` so
```
 tls_certificate_path: /config/whatevercertificate.crt 
 tls_private_key_path: /config/whatevercertificate.key 
```

**Reply by XenyFord** [reply]
Mr. Will,

Could you please be so kind to help me on how to create the certificate and keys. FYI: my postal site is still showing a secured connection lock as I installed it using Docker with Caddy by following the recent documentation published on the official portal. Which means the SSL/TLS is available somewhere on the server?

Await your valuable response. Thank you in advance!

![secure](https://user-images.githubusercontent.com/89290310/132016192-13787408-c628-484b-ad08-ba88f50bd97d.jpg)

**Reply by hswamy** [reply]
Resolved my issue

**Reply by willpower232** [reply]
you may be able to get a copy of the certificates from caddy but I don't know how to do that.
